runtime/doc/map.txt

```diff
@@ -1431,6 +1431,41 @@ Possible values are (second column is the short name used in listing):
     -addr=other		  ?	other kind of range
 
 
+Incremental preview ~
+                                                  *:command-preview*
+It's possible to allow commands to be incrementally previewed when
+'inccommand' is enabled by using the -preview command attribute. Possible use
+cases are:
+	-preview		Always preview
+	-preview={func}		Preview only if {func} returns |TRUE|
+
+The {func} part must be a function with the following signature: >
+
+	:function {func}(CmdLine)
+
+The function is only given one argument, which is the entire command line. The
+function must return a boolean value which determines whether the command
+preview should be triggered or not. If -preview is used by itself without any
+arguments, the command is always previewed whenever the command line is
+modified.
+
+Note that every time a command needs to be previewed, Nvim executes the
+command. It's possible to check if a command is being executed for
+previewing by using the "cmdpreview" key of |nvim_get_mode()|.
+
+Here is an example that uses the -preview attribute to preview a command that
+trims leading whitespaces from lines: >
+
+	function! PreviewFunc(cmd)
+	    " Preview when first argument equals 'enable' (case-insensitive)
+	    let args = split(a:cmd, ' ')
+	    return len(args) > 1 && args[1] ==? 'enable'
+	endfunction
+
+	command! -nargs=1 -preview=PreviewFunc -range=% -addr=lines
```

Comment:
Don't know if we want to extend the vimscript/ex-mode implementation this has been avoided in the past (https://github.com/neovim/neovim/pull/17426) favoring implementation on the API level.

Reply:
You mean the -preview attribute? I think we should make an exception in this case personally, but I'm free to make changes if required